content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -1228,38 +1231,46 @@ private void handleSubmissionStatus(String submissionStatus, ContentReviewItem i
 			case "CREATED":
 				// do nothing... try again
 				break;
-			case "UNSUPPORTED_FILETYPE":
-				errorStr = "The uploaded filetype is not supported";
-				break;
-				//break on all
-			case "PROCESSING_ERROR":
-				errorStr = "An unspecified error occurred while processing the submissions";
-				break;
-			case "TOO_LITTLE_TEXT":
-				errorStr = "The submission does not have enough text to generate a Similarity Report (a submission must contain at least 20 words)";
-				break;
-			case "TOO_MUCH_TEXT":
-				errorStr = "The submission has too much text to generate a Similarity Report (after extracted text is converted to UTF-8, the submission must contain less than 2MB of text)";
-				break;
-			case "TOO_MANY_PAGES":
-				errorStr = "The submission has too many pages to generate a Similarity Report (a submission cannot contain more than 400 pages)";
-				break;
-			case "FILE_LOCKED":
-				errorStr = "The uploaded file requires a password in order to be opened";
-				break;
-			case "CORRUPT_FILE":
-				errorStr = "The uploaded file appears to be corrupt or password protected";
-				break;
-			case "ERROR":
-				errorStr = "Submission returned with ERROR status";
-				break;
 			default:
-				log.info("Unknown submission status, will retry: " + submissionStatus);
-				break;
+				String errorCode = submissionJSON.containsKey("error_code") ? submissionJSON.getString("error_code") : submissionStatus;
+				switch (errorCode)
+				{
+				case "UNSUPPORTED_FILETYPE":
+					errorStr = "The uploaded filetype is not supported";
+					break;
+					//break on all
+				case "PROCESSING_ERROR":
+					errorStr = "An unspecified error occurred while processing the submissions";
```

Comment:
Are these error strings user facing? If so they should be internationalized.

Reply:
Yes, as a result of SAK-41791. This is ubiquitous to every content review impl right now. I was thinking a ticket would be warranted to go through all of them and internationalize every value that gets set on the 'last_error' column

Reply:
Will do this next in my queue. See SAK-41883